### ISSUE TITLE
optimize/convex/lp: `panic: blas: index of a out of range` macOS High Sierra

### ISSUE THREAD

**Issue by jjhbw** [gonum/gonum#441]
Thank you guys for making this awesome library!

I got a BLAS-related panic when running `lp.Simplex` on a tiny linear optimization problem. 
I am working on macOS High Sierra `10.13.3 (17D102)`. 

I am not really at home working directly with BLAS, but let me know if you need any additional information.

### What are you trying to do?
Solve a linear program using `lp.Simplex`.

### What did you do?
```go
func Test_ThisBreaksGonumSimplex(t *testing.T) {
	c := []float64{-1, -2}
	A := mat.NewDense(2, 2, []float64{
		-1, 2.6,
		3, 1.1,
	})
	b := []float64{4, 9}

	lp.Simplex(c, A, b, 0, nil)
}

```

### What did you expect to happen?
I expected a solution or an error letting me know my problem is formulated wrong.

### What actually happened?
I got the following `panic`.
```
--- FAIL: Test_ThisBreaksGonumSimplex (0.00s)
panic: blas: index of a out of range [recovered]
	panic: blas: index of a out of range

goroutine 6 [running]:
testing.tRunner.func1(0xc4201000f0)
	/usr/local/Cellar/go/1.9.2/libexec/src/testing/testing.go:711 +0x2d2
panic(0x13271c0, 0x13d25e0)
	/usr/local/Cellar/go/1.9.2/libexec/src/runtime/panic.go:491 +0x283
gonum.org/v1/gonum/blas/gonum.Implementation.Dgemv(0x70, 0x2, 0x0, 0x3ff0000000000000, 0x156a818, 0x0, 0x0, 0x0, 0xc4201040a0, 0x2, ...)
	/Users/jur/go/src/gonum.org/v1/gonum/blas/gonum/level2double.go:29 +0x975
gonum.org/v1/gonum/blas/gonum.(*Implementation).Dgemv(0x156a580, 0x70, 0x2, 0x0, 0x3ff0000000000000, 0x156a818, 0x0, 0x0, 0x0, 0xc4201040a0, ...)
	<autogenerated>:1 +0x135
gonum.org/v1/gonum/blas/blas64.Gemv(0x70, 0x3ff0000000000000, 0x2, 0x0, 0x0, 0x156a818, 0x0, 0x0, 0x1, 0xc4201040a0, ...)
	/Users/jur/go/src/gonum.org/v1/gonum/blas/blas64/blas64.go:215 +0x12e
gonum.org/v1/gonum/mat.(*VecDense).MulVec(0xc4200110b0, 0x1514c40, 0xc420046e80, 0x1515e00, 0xc420011080)
	/Users/jur/go/src/gonum.org/v1/gonum/mat/vector.go:608 +0xc76
gonum.org/v1/gonum/optimize/convex/lp.simplex(0x0, 0x0, 0x0, 0xc420057f78, 0x2, 0x2, 0x1514980, 0xc4200607c0, 0xc420019fe0, 0x2, ...)
	/Users/jur/go/src/gonum.org/v1/gonum/optimize/convex/lp/simplex.go:224 +0xe25
gonum.org/v1/gonum/optimize/convex/lp.Simplex(0xc420032778, 0x2, 0x2, 0x1514980, 0xc4200607c0, 0xc420019fe0, 0x2, 0x2, 0x0, 0x0, ...)
	/Users/jur/go/src/gonum.org/v1/gonum/optimize/convex/lp/simplex.go:89 +0xd1
gitlab.com/jjhbarkeywolf/ilp.Test_ThisBreaksGonumSimplex(0xc4201000f0)
	/Users/jur/go/src/gitlab.com/jjhbarkeywolf/ilp/ilp_test.go:663 +0x151
testing.tRunner(0xc4201000f0, 0x13a9fb0)
	/usr/local/Cellar/go/1.9.2/libexec/src/testing/testing.go:746 +0xd0
created by testing.(*T).Run
	/usr/local/Cellar/go/1.9.2/libexec/src/testing/testing.go:789 +0x2de
FAIL	gitlab.com/jjhbarkeywolf/ilp	0.018s
Error: Tests failed.
```

### What version of Go and Gonum are you using?

```
go version go1.9.2 darwin/amd64
```
```
(cd $(go env GOPATH)/src/gonum.org/v1/gonum && git rev-parse HEAD)
052ff404ed7bcbcf5c326f45dd2e8021cef6af9f
```

### Does this issue reproduce with the current master?
Yes. Used version is current master commit.

**Comment by btracey** [comment]
Thanks for the report. This is triggered because the LP has the same number of constraints as variables, and so there are no degrees of freedom.

**Comment by jjhbw** [comment]
Ah, that explains why problems like these trigger the edge case. I was auto-generating problems, you see, and a bunch of them started failing like this. Now I see the pattern, thanks!

**Comment by btracey** [comment]
Fixed by #443
